dspace-server-webapp/src/main/java/org/dspace/app/rest/DiscoveryRestController.java

```diff
@@ -149,13 +152,22 @@ public SearchResultsResource getSearchObjects(@RequestParam(name = "query", requ
         }
 
         //Get the Search results in JSON format
-        SearchResultsRest searchResultsRest = discoveryRestRepository
-            .getSearchObjects(query, dsoTypes, dsoScope, configuration, searchFilters, page, utils.obtainProjection());
-
-        //Convert the Search JSON results to paginated HAL resources
-        SearchResultsResource searchResultsResource = new SearchResultsResource(searchResultsRest, utils, page);
-        halLinkService.addLinks(searchResultsResource, page);
-        return searchResultsResource;
+        try {
+            SearchResultsRest searchResultsRest = discoveryRestRepository.getSearchObjects(query, dsoTypes, dsoScope,
+                configuration, searchFilters, page, utils.obtainProjection());
+
+            //Convert the Search JSON results to paginated HAL resources
+            SearchResultsResource searchResultsResource = new SearchResultsResource(searchResultsRest, utils, page);
+            halLinkService.addLinks(searchResultsResource, page);
+            return searchResultsResource;
+        } catch (IllegalArgumentException e) {
+            boolean isParsingException = e.getMessage().contains(SOLR_PARSE_ERROR_MESSAGE);
```

Comment:
I'm worried that this is a fragile way of determining whether Solr threw a SyntaxError. Usually, we don't try to depend on the text of an error message to determine the error type (as if Solr changes the error message here, our code will start to strangely fail)

Are you sure this is the only way to achieve this?  It looks, to me, like the `IllegalArgumentException` thrown here is wrapping a `org.apache.solr.search.SyntaxError` (which is what Solr actually throws).  Maybe we could just check what error we are wrapping, somewhat like this https://stackoverflow.com/a/29252884

```
if (ExceptionUtils.indexOfThrowable(e, SyntaxError.class) != 1) {
    throw new UnprocessableEntityException(e.getMessage());
}
```

That might remove the need for this `SOLR_PARSE_ERROR_MESSAGE` constant altogether.

Reply:
@tdonohue I do agree that this solution is not robust but I haven't been able to find a better solution.

I cannot use `ExceptionUtils` because that specific exception is thrown by Solr server, not by DSpace. The root exception in the stack trace is:

```
Caused by: org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteSolrException: Error from server at http://localhost:8983/solr/search: org.apache.solr.search.SyntaxError: Cannot parse 'test:': Encountered "<EOF>" at line 1, column 5.
```

The only Java exception that can be handled with `ExceptionUtils` is `RemoteSolrException`, which is too much generic, and `org.apache.solr.search.SyntaxError` is just plain text in the error message. However, I could replace the value of `SOLR_PARSE_ERROR_MESSAGE` with the name of the exception. This could make the code slightly less non-robust, assuming that the exception name is more unlikely to be changed rather than exception message.

Reply:
@davidenegretti-4science : OK, in that case, I'd recommend we change the value of `SOLR_PARSE_ERROR_MESSAGE` to include the name of the exception, as you suggested.  I'd also ask that you add some inline code comments to describe *why* we are doing this approach (as it's not very robust & I don't want to recommend that other developers "borrow" or reuse this code elsewhere in future work).  So, even saying something like:

"We unfortunately have to do a string comparison to locate the source of the error, as Solr only sends back a generic exception, and the org.apache.solr.search.SyntaxError is only available in the error message."

Reply:
@tdonohue I've pushed this correction